backend/packages/Upgrade/src/api/services/QueryService.ts

```diff
@@ -34,13 +37,35 @@ export class QueryService {
 
     const promiseResult = await Promise.all(promiseArray);
     const analysePromise = promiseResult.map((query) => this.logRepository.analysis(query));
-    let response = await Promise.all(analysePromise);
-    response = response.map((res, index) => {
-      return {
-        id: queryIds[index],
-        result: res,
-      };
+    const response = await Promise.allSettled(analysePromise);
+
+    const failedQuery = Array<Promise<any>>();
+
+    let modifiedResponse = response.map((query, index) => {
+      if (query.status === 'fulfilled') {
+        return query.value;
+      } else {
+        this.log.error('Error in Query Id ', queryIds[index]);
+        failedQuery.push(this.errorService.create({
+          endPoint: '/api/query/analyse',
+          errorCode: 500,
+          message: `Query Failed error: ${JSON.stringify(queryIds[index], undefined, 2)}`,
+          name: 'Query Failed error',
+          type: SERVER_ERROR.QUERY_FAILED,
+        } as any));
+
+        return [];
+      }
     });
-    return response;
+
+    if (failedQuery.length) {
+      await Promise.all(failedQuery);
+    }
+
+    modifiedResponse = modifiedResponse.map((res, index) => {
+      return queryIds[index] ? { id: queryIds[index], result: res} : null;
```

Comment:
I think. the condition `queryIds[index]` will always have value. In what can null value will be returned to map function?  Also fix the formatting issue.

Reply:
modifiedResponse had empty array as response (in case the promise was rejected). I have updated the code wherein the response is first filtered so we won't have empty array.